cookbooks/cdo-varnish/templates/default/accept-language.vcl.erb

```diff
@@ -0,0 +1,188 @@
+C{
+
+/* ------------------------------------------------------------------ */
+/* THIS FILE IS AUTOMATICALLY GENERATED BY ./gen_vcl.pl. DO NOT EDIT. */
+
+/*
+ * Accept-language header normalization
+ *
+ * - Parses client Accept-Language HTTP header
+ * - Tries to find the best match with the supported languages
+ * - Writes the best match as req.http.X-Varnish-Accept-Language
+ *
+ * First version: Cosimo, 21/Jan/2010
+ * Last update:   Cosimo, 03/Nov/2011
+ *
+ * http://github.com/cosimo/varnish-accept-language
+ *
+ */
+
+#include <ctype.h>  /* isupper */
+#include <stdio.h>
+#include <stdlib.h> /* qsort */
+#include <string.h>
+
+#define DEFAULT_LANGUAGE "en"
+#define SUPPORTED_LANGUAGES ":<%= node['cdo-apps']['i18n']['languages'].to_hash.keys.join(':') %>:"
+
+#define vcl_string char
+#define LANG_LIST_SIZE 16
+#define HDR_MAXLEN 256
+#define LANG_MAXLEN 8
+#define RETURN_LANG(x) { \
```

Comment:
If this code is safe as long as x is not longer than `LANG_MAXLEN` (8), then it should be safe as long as none of the keys in our [list of whitelisted languages](https://github.com/code-dot-org/code-dot-org/blob/staging/cookbooks/cdo-varnish/files/default/languages.json) exceeds this length (max is currently 5), does that sound right?